server/src/main/java/org/apache/iotdb/db/engine/compaction/performer/impl/ReadPointCompactionPerformer.java

```diff
@@ -146,19 +153,9 @@ private void compactAlignedSeries(
     MultiTsFileDeviceIterator.AlignedMeasurementIterator alignedMeasurementIterator =
         deviceIterator.iterateAlignedSeries(device);
     Set<String> allMeasurements = alignedMeasurementIterator.getAllMeasurements();
-    List<IMeasurementSchema> measurementSchemas = new ArrayList<>();
-    for (String measurement : allMeasurements) {
-      try {
-        if (IoTDBDescriptor.getInstance().getConfig().isEnableIDTable()) {
-          measurementSchemas.add(IDTableManager.getInstance().getSeriesSchema(device, measurement));
-        } else {
-          measurementSchemas.add(
-              IoTDB.schemaProcessor.getSeriesSchema(new PartialPath(device, measurement)));
-        }
-      } catch (PathNotExistException e) {
-        LOGGER.info("A deleted path is skipped: {}", e.getMessage());
-      }
-    }
+    ConcurrentHashMap<String, MeasurementSchema> schemaMap =
+        getMeasurementSchema(device, allMeasurements);
+    List<IMeasurementSchema> measurementSchemas = new ArrayList<>(schemaMap.values());
```

Comment:
The map is read-only after initialization, so it does not seem necessary to use a concurrent one.